src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/NullableReferenceTypesRocks.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Java.Interop.Tools.Cecil
+{
+	// Partial support for determining NRT status of method and field types.
+	// Reference: https://github.com/dotnet/roslyn/blob/main/docs/features/nullable-metadata.md
+	// The basics are supported, but advanced annotations like array elements,
+	// type parameters, and tuples are not supported.  Our use case doesn't really need them.
+	public static class NullableReferenceTypesRocks
+	{
+		public static Nullability GetTypeNullability (this FieldDefinition field)
+		{
+			if (field.FieldType.FullName == "System.Void")
```

Comment:
A "concern" here is that this doesn't check for assembly name, so if any yahoo adds `System.Void` to their assembly, it will be treated the same as `System.Void, mscorlib` or `System.Void, System.Private.CoreLib` or `System.Void, netstandard`.

I'm not sure of a good way to say "this is a core assembly", so this concern is likely ignorable, but it is something to be aware of.

Reply:
I think we can live with this.  I'm not comfortable trying to enumerate every "core" assembly that they might have put `System.Void` in, or will in the future.  😉